### PR TITLE
Milestone 2: Add and Remove items

### PR DESCRIPTION
**Pull request by Xghayor**
In this milestone i completed:
1.Added add function to add items.
2.Added remove function to remove items.
3.Added styles to the page

### CODE REVIEW

File: src/removeTasks.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+document.addEventListener('click', (event) => {
+  const clickedElement = event.target;
+  if (clickedElement.classList.contains('checking-box')) {
+    const listItem = clickedElement.parentNode;
+    const index = Array.from(listItem.parentNode.children).indexOf(listItem);
+    if (index !== -1) {
+      removeTask(index);
+      createItems();
+    }
+  }
```
Comment (V-Blaze) [inline src/removeTasks.js]: - You have done great so far, but there are some changes you need to make before moving to the next milestone. :+1: :+1: 
> Implement a function for deleting a task (remove an element from the array).

- [ ] You tried implementing the remove task functionality but your approach doesn't meet the project requirement, From your current implementation, a user needs to mark the checkbox and then click the `clear completed` to delete a task, but that's not the flow. You need to add a delete icon in front of each individual task and add an `onclick event listener` to delete the specific task. I'll suggest you implement this. :+1: :100:

File: src/index.js [hunk 2]
```diff
@@ -1,6 +1,29 @@
...
+});
+
+const addButton = document.querySelector('.add-task button');
+const inputField = document.getElementById('add-tasks');
+
+addButton.addEventListener('click', addTask);
+inputField.addEventListener('keydown', (event) => {
+  if (event.key === 'Enter') {
+    addTask();
+  }
```
Comment (V-Blaze) [inline src/index.js]: > Implement a function for editing task descriptions.
- [ ] From the project requirement, you are meant to also implement the `task editing` feature in this milestone. I'll suggest you implement this in your project. :+1: :+1:

File: src/taskManager.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    completed: false,
+    index: tasks.length + 1,
+  };
+
+  tasks.push(task);
+}
+
+function removeTask(index) {
+  tasks.splice(index, 1);
+}
```
Comment (V-Blaze) [inline src/taskManager.js]: > Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1)..
- [ ] According to the project requirement, deleting a task should rearrange the indexes of the previous tasks. Currently, you haven't implanted this in your code. I'll suggest you write out a function that does this and call the function after deleting each task. :+1: :smile:

File: src/taskManager.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
+// eslint-disable-next-line import/no-mutable-exports
```
Comment (V-Blaze) [inline src/taskManager.js]: - [ ] As required by Microverse, you are not meant to use any linters diable rule in your codebase. The linters are there to make sure your code meets the best standard practices. I'll suggest you remove this linter disable rule and fix the issue. :+1: :+1: :+1:

File: src/addItem.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    return;
+  }
+  TaskManager.addTask(description);
+  TaskManager.saveTasks();
+
+  inputField.value = '';
+
+  createItems();
+}
+// eslint-disable-next-line import/prefer-default-export
```
Comment (V-Blaze) [inline src/addItem.js]: - [ ] As i have suggested above, you are not meant to use any linters diable rule in your codebase. The linters are there to make sure your code meets the best standard practices. I'll suggest you remove this linter diable rule and fix the issue. :+1: :+1: :+1:

File: src/createItems.js [hunk 6]
```diff
@@ -26,20 +20,27 @@ function createItems(tasks) {
...
-      listItem.classList.add('completed');
-    }
     todoList.appendChild(listItem);
 
     const hr = document.createElement('hr');
     todoList.appendChild(hr);
   });
 }
-export { createItems, tasks };
+// eslint-disable-next-line import/prefer-default-export
```
Comment (V-Blaze) [inline src/createItems.js]: - [ ] As i have suggested above, you are not meant to use any linters diable rule in your codebase. The linters are there to make sure your code meets the best standard practices. I'll suggest you remove this linter diable rule and fix the issue. :+1: :+1: :+1:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @Xghayor , 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Adding task works great✔
-  tasks are stored and fetched from local storage👋 
-  correct Github flow✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lizdev-05 — APPROVED** [review]
Hi @Xghayor , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

:heavy_check_mark: Professional Readme file :+1: 
:heavy_check_mark: Good commit history :+1:
:heavy_check_mark: No linter error :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**